pkg/service-manager/clients/k8s_service_catalog.go

```diff
@@ -0,0 +1,436 @@
+				if ps != nil && ps.Size() > 0 {
+					b, _ := json.Marshal(ps)
+					if err := json.Unmarshal(b, out); err != nil {
+						return nil, errors.Wrap(err, "could not decode schema")
+					}
+				}
+				return out, nil
+			}
+			create, err := marshall(plan.Spec.ServiceInstanceCreateParameterSchema)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error marshalling create schema")
+			}
+			update, err := marshall(plan.Spec.ServiceInstanceUpdateParameterSchema)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error marshalling update schema")
+			}
+			bind, err := marshall(plan.Spec.ServiceBindingCreateParameterSchema)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error marshalling bind schema")
+			}
+
+			bindable := csc.Spec.Bindable
+			if plan.Spec.Bindable != nil {
+				bindable = *plan.Spec.Bindable
+			}
+
+			serviceClassPlans = append(serviceClassPlans, entities.ServicePlan{
+				BaseEntity: entitystore.BaseEntity{
+					Name:   plan.Spec.ExternalName,
+					Status: entitystore.StatusREADY,
+				},
+				PlanID:      plan.Name,
+				Description: plan.Spec.Description,
+				Schema: entities.Schema{
+					Create: create,
+					Update: update,
+					Bind:   bind,
+				},
+				Free:     plan.Spec.Free,
+				Bindable: bindable,
+				Metadata: plan.Spec.ExternalMetadata,
+			})
+		}
+		serviceClasses = append(serviceClasses, &entities.ServiceClass{
+			BaseEntity: entitystore.BaseEntity{
+				Name:   csc.Spec.ExternalName,
+				Status: entitystore.StatusREADY,
+			},
+			ServiceID: csc.Name,
+			Broker:    csc.Spec.ClusterServiceBrokerName,
+			Bindable:  csc.Spec.Bindable,
+			Plans:     serviceClassPlans,
+		})
+	}
+
+	return serviceClasses, nil
+}
+
+// ListServiceInstaces returns a list of ServiceInstance entities which correspond to the provisioned services.  Most
+// importantly the status of each service instance is returned as well.
+func (c *k8sServiceCatalogClient) ListServiceInstances() ([]entitystore.Entity, error) {
+	instances, err := c.sdk.RetrieveInstances(c.config.CatalogNamespace)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error retreiving service instance list")
+	}
+	var serviceInstances []entitystore.Entity
+	for _, instance := range instances.Items {
+
+		parameters := make(map[string]interface{})
+		err := json.Unmarshal(instance.Spec.Parameters.Raw, &parameters)
+		if err != nil {
+			log.Errorf("Failed to decode the service instance parameters for %s", instance.ObjectMeta.Name)
+		}
+		serviceInstance := &entities.ServiceInstance{
+			BaseEntity: entitystore.BaseEntity{
+				ID:     instance.ObjectMeta.Name,
+				Status: entitystore.StatusUNKNOWN,
+			},
+			ServiceClass: instance.Spec.ClusterServiceClassExternalName,
+			ServicePlan:  instance.Spec.ClusterServicePlanExternalName,
+			Parameters:   parameters,
+		}
+		for _, cond := range instance.Status.Conditions {
+			if cond.Type == v1beta1.ServiceInstanceConditionReady && cond.Status == v1beta1.ConditionTrue {
```

Comment:
We should also handle the case when `cond.Type == v1beta1.ServiceInstanceConditionReady && cond.Status == v1beta1.ConditionFalse`? 

https://github.com/kubernetes-incubator/service-catalog/blob/f7c622f6491fc9155871dcd75cec4c4df006a418/pkg/controller/controller_instance.go#L996

Reply:
Wow, you are thorough!  Also, thanks!